### PR TITLE
Refactor min/maxReplicas

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- Move the `minReplicas` and the `maxReplicas` to the separated obj. `scaleConfig`.

**Related issue(s)**

* https://github.com/kyma-project/kyma/pull/14234
* https://github.com/kyma-project/kyma/issues/14593#issuecomment-1185289026

### CODE REVIEW

File: components/function-controller/internal/webhook/conversion_webhook.go [hunk 1]
```diff
@@ -277,10 +281,12 @@ func (w *ConvertingWebhook) convertFunctionV1Alpha2ToV1Alpha1(src, dst runtime.O
 
 func (w *ConvertingWebhook) convertSpecV1Alpha2ToV1Alpha1(in *serverlessv1alpha2.Function, out *serverlessv1alpha1.Function) error {
 	out.Spec.Env = in.Spec.Env
-	out.Spec.MaxReplicas = in.Spec.MaxReplicas
-	out.Spec.MinReplicas = in.Spec.MaxReplicas
```
Comment (pPrecel) [inline components/function-controller/internal/webhook/conversion_webhook.go:281]: It looks like a bug when the wrong field is assigned to minReplicas (should be min -> min, not max -> min) so I decided to change it. Am I right?

**Review by Cortey — APPROVED** [review]

**Review by tobiscr — DISMISSED** [review]

**Review by tobiscr — APPROVED** [review]

**Review by Cortey — APPROVED** [review]

**Pull request merged**